src/Console/Commands/MvcUpdateContract.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Tychovbh\Mvc\Console\Commands;
+
+use Illuminate\Console\Command;
+use Tychovbh\Mvc\Contract;
+use Tychovbh\Mvc\Services\DocumentSign\DocumentSignInterface;
+
+class MvcUpdateContract extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'mvc:contracts-update';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'This command updates the signed_at from a contract, when the status is si(signed)';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @param DocumentSignInterface $documentSign
+     * @return mixed
+     */
+    public function handle(DocumentSignInterface $documentSign)
+    {
+        try {
+            $contracts = Contract::all()->whereNotNull('external_id')->whereNull('signed_at');
```

Comment:
this is a waste you are retrieving all records and than you are filtering those records. Always use a direct query
Contract::whereNotnull()->get();